src/huggingface_hub/utils/_errors.py

```diff
@@ -0,0 +1,41 @@
+from requests import HTTPError
+
+
+class RepositoryNotFoundError(HTTPError):
```

Comment:
These exceptions need to be added to the list of documented classes since users will be exposed to them.

Reply:
Added them, and mentionned they could be raised in all places